Agent start fails with nil pointer dereference

Current master branch: 104b12040a8fbe6153d1a4f1be1b73d37bf699e5

% ~/repo/consul/bin/consul agent -dev -advertise 127.0.0.1 -config-dir=etc/consul.d
==> Starting Consul agent...
==> Starting Consul agent RPC...
==> Consul agent running!
           Version: 'v0.7.0-111-g104b120-dev (104b120)'
         Node name: 'LM-AMS-14500281'
        Datacenter: 'dc1'
            Server: true (bootstrap: false)
       Client Addr: 127.0.0.1 (HTTP: 8500, HTTPS: -1, DNS: 8600, RPC: 8400)
      Cluster Addr: 127.0.0.1 (LAN: 8301, WAN: 8302)
    Gossip encrypt: false, RPC-TLS: false, TLS-Incoming: false
             Atlas: <disabled>

==> Log data will now stream in as it occurs:

```
2016/10/21 16:00:22 [INFO] raft: Initial configuration (index=1): [{Suffrage:Voter ID:127.0.0.1:8300 Address:127.0.0.1:8300}]
2016/10/21 16:00:22 [INFO] raft: Node at 127.0.0.1:8300 [Follower] entering Follower state (Leader: "")
2016/10/21 16:00:22 [INFO] serf: EventMemberJoin: LM-AMS-14500281 127.0.0.1
2016/10/21 16:00:22 [INFO] consul: Adding LAN server LM-AMS-14500281 (Addr: tcp/127.0.0.1:8300) (DC: dc1)
2016/10/21 16:00:22 [INFO] serf: EventMemberJoin: LM-AMS-14500281.dc1 127.0.0.1
2016/10/21 16:00:22 [INFO] consul: Adding WAN server LM-AMS-14500281.dc1 (Addr: tcp/127.0.0.1:8300) (DC: dc1)
2016/10/21 16:00:22 [DEBUG] agent: pausing 50.098827ms before first invocation of redis-cli -p 6379 ping || exit 1
2016/10/21 16:00:22 [DEBUG] agent: pausing 957.557913ms before first invocation of redis-cli -p 6381 ping || exit 1
2016/10/21 16:00:22 [DEBUG] agent: pausing 819.904891ms before first invocation of redis-cli -p 6380 ping || exit 1
```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x140782]

goroutine 72 [running]:
panic(0x6f4720, 0xc420012100)
    /usr/local/go/src/runtime/panic.go:500 +0x1a1
sync.(_RWMutex).RLock(0x0)
    /usr/local/go/src/sync/rwmutex.go:41 +0x22
github.com/hashicorp/consul/command/agent.(_CheckMonitor).check(0xc42025c770)
    /Users/pdekloe/repo/consul/src/github.com/hashicorp/consul/command/agent/check.go:160 +0x62
github.com/hashicorp/consul/command/agent.(_CheckMonitor).run(0xc42025c770)
    /Users/pdekloe/repo/consul/src/github.com/hashicorp/consul/command/agent/check.go:147 +0x1d0
created by github.com/hashicorp/consul/command/agent.(_CheckMonitor).Start
    /Users/pdekloe/repo/consul/src/github.com/hashicorp/consul/command/agent/check.go:125 +0xbf
%

--- a/command/agent/check.go
+++ b/command/agent/check.go
@@ -110,7 +110,6 @@ type CheckMonitor struct {
 	Interval time.Duration
 	Timeout  time.Duration
 	Logger   *log.Logger
-	ReapLock *sync.RWMutex
 
 	stop     bool
 	stopCh   chan struct{}
@@ -156,12 +155,6 @@ func (c *CheckMonitor) run() {
 
 // check is invoked periodically to perform the script check
 func (c *CheckMonitor) check() {
-	// Disable child process reaping so that we can get this command's
-	// return value. Note that we take the read lock here since we are
-	// waiting on a specific PID and don't need to serialize all waits.
-	c.ReapLock.RLock()
-	defer c.ReapLock.RUnlock()
-
 	// Create the command
 	cmd, err := ExecScript(c.Script)
 	if err != nil {

--- a/command/agent/check_test.go
+++ b/command/agent/check_test.go
@@ -11,7 +11,6 @@ import (
 	"os"
 	"os/exec"
 	"strings"
-	"sync"
 	"testing"
 	"time"
 
@@ -46,7 +45,6 @@ func expectStatus(t *testing.T, script, status string) {
 		Script:   script,
 		Interval: 10 * time.Millisecond,
 		Logger:   log.New(os.Stderr, "", log.LstdFlags),
-		ReapLock: &sync.RWMutex{},
 	}
 	check.Start()
 	defer check.Stop()
@@ -96,7 +94,6 @@ func TestCheckMonitor_Timeout(t *testing.T) {
 		Interval: 10 * time.Millisecond,
 		Timeout:  5 * time.Millisecond,
 		Logger:   log.New(os.Stderr, "", log.LstdFlags),
-		ReapLock: &sync.RWMutex{},
 	}
 	check.Start()
 	defer check.Stop()
@@ -125,7 +122,6 @@ func TestCheckMonitor_RandomStagger(t *testing.T) {
 		Script:   "exit 0",
 		Interval: 25 * time.Millisecond,
 		Logger:   log.New(os.Stderr, "", log.LstdFlags),
-		ReapLock: &sync.RWMutex{},
 	}
 	check.Start()
 	defer check.Stop()
@@ -154,7 +150,6 @@ func TestCheckMonitor_LimitOutput(t *testing.T) {
 		Script:   "od -N 81920 /dev/urandom",
 		Interval: 25 * time.Millisecond,
 		Logger:   log.New(os.Stderr, "", log.LstdFlags),
-		ReapLock: &sync.RWMutex{},
 	}
 	check.Start()
 	defer check.Stop()

